opentelemetry/proto/trace/v1/trace.proto

```diff
@@ -24,6 +24,25 @@ option java_package = "io.opentelemetry.proto.trace.v1";
 option java_outer_classname = "TraceProto";
 option go_package = "github.com/open-telemetry/opentelemetry-proto/gen/go/trace/v1";
 
+// Traces represents the traces data that can be stored in a persistent storage,
+// OR can be embedded by other protocols that transfer OTLP traces data but do
+// not implement the OTLP protocol.
+//
+// The main difference between this message and ExportTraceServiceRequest is that
+// in this message there will not be any "control" or "metadata" specific to
+// OTLP protocol.
+//
+// When new fields are added into this message, ExportTraceServiceRequest MUST be updated
+// as well.
+message Traces {
```

Comment:
Is it ok that a structure `Traces` will generally not contain full traces, just the part that is local to a component? It seems like it could be confusing (`ExportTraceRequest` doesn't have this problem since it's explicitly about exporting).

Reply:
Indeed it may be a bit confusing, but on the other side I really like the consistency.

Maybe rename everything with a suffix 'Data'?

Reply:
`Data` came to mind to and might be OK. Though the `List` suffix seems dead simple and pretty clear too

Reply:
Yeah, `List` or `Collection` or similar should work.

Reply:
I like `List`.

Reply:
ResourceSpansList pe TracesList? Not sure I understand the proposal exactly

Reply:
TracesList sounds like it is full traces like @anuraaga said. ResourceSpansList seems safer.

Reply:
ResourceSpansList is bad because the whole idea is that we may possibly add extra things. Naming very specific like that is not my top preference

Reply:
Picked "TracesData" since again see my previous comment, one of the reason to have this as independent message is to ensure that people embedding this message will have all the "data" fields that a signal may have, excluding protocol specific metadata.